device_coupler/runner.py

```diff
@@ -0,0 +1,110 @@
+"""Module to run device coupler"""
+
+from __future__ import absolute_import
+
+from proto import system_config_pb2 as sys_config
+from daq.proto.device_coupler_pb2 import DiscoveryEventType
+
+from device_coupler.utils import get_logger, yaml_proto
+from device_coupler.network_helper import NetworkHelper
+from device_coupler.device_discovery import DeviceDiscovery
+from device_coupler.daq_client import DAQClient
+from device_coupler.ovs_helper import OvsHelper
+
+from queue import Queue, Empty
+
+import time
+import threading
+
+
+class DeviceCoupler():
+    """Main for device coupler"""
+
+    _WORKER_TIMEOUT = 10
+    _WORKER_COUNT = 3
+
+    def __init__(self, config):
+        self._logger = get_logger('device_coupler')
+        self._config = config
+        self._test_vlans = list(config.test_vlans)
+        self._network_helper = None
+        self._device_discovery = None
+        self._daq_client = None
+        self._event_queue = None
+        self._running = None
+        self._workers = None
+        self._source_ip = None
+
+    def setup(self):
+        """Setup device coupler"""
+        self._running = True
+        self._ovs_helper = OvsHelper()
+
+        self._network_helper = NetworkHelper(
+            self._config.trunk_iface, self._config.bridge, self._test_vlans)
+        self._network_helper.setup()
+
+        self._event_queue = Queue()
+        self._build_worker_threads(self._process_event_queue, self._WORKER_COUNT)
```

Comment:
I think you can simplify these by using a ThreadPoolExecutor here for process_event_queue and polling of the forwarding table.

Reply:
I figured ThreadPoolExecutor might be overkill for what I Was doing. Replaced it with ThreadPoolExecutor though.